scripts/deforum_helpers/animation_key_frames.py

```diff
@@ -5,105 +5,113 @@
-    key_frame_series = key_frame_series.interpolate(method=interp_method.lower(), limit_direction='both')
-    if integer:
-        return key_frame_series.astype(int)
-    return key_frame_series
+                # workaround for values formatted like 0:("I am test") //used for sampler schedules
+                key_frame_series[i] = numexpr.evaluate(value) if not is_single_string else self.sanitize_value(value)
+            elif is_single_string:# take previous string value and replicate it
+                key_frame_series[i] = key_frame_series[i-1]
+        key_frame_series = key_frame_series.astype(float) if not is_single_string else key_frame_series # as string
+        
+        if interp_method == 'Cubic' and len(key_frames.items()) <= 3:
+            interp_method = 'Quadratic'    
+        if interp_method == 'Quadratic' and len(key_frames.items()) <= 2:
+            interp_method = 'Linear'
+            
+        key_frame_series[0] = key_frame_series[key_frame_series.first_valid_index()]
+        key_frame_series[self.max_frames-1] = key_frame_series[key_frame_series.last_valid_index()]
+        key_frame_series = key_frame_series.interpolate(method=interp_method.lower(), limit_direction='both')
+        if integer:
+            return key_frame_series.astype(int)
+        return key_frame_series
 
-def parse_key_frames(string, prompt_parser=None):
-    # because math functions (i.e. sin(t)) can utilize brackets 
-    # it extracts the value in form of some stuff
-    # which has previously been enclosed with brackets and
-    # with a comma or end of line existing after the closing one
-    pattern = r'((?P<frame>[0-9]+):[\s]*\((?P<param>[\S\s]*?)\)([,][\s]?|[\s]?$))'
-    frames = dict()
-    for match_object in re.finditer(pattern, string):
-        frame = int(match_object.groupdict()['frame'])
-        param = match_object.groupdict()['param']
-        if prompt_parser:
-            frames[frame] = prompt_parser(param)
-        else:
-            frames[frame] = param
-    if frames == {} and len(string) != 0:
-        raise RuntimeError('Key Frame string not correctly formatted')
-    return frames
\ No newline at end of file
+    def parse_key_frames(self, string):
+        # because math functions (i.e. sin(t)) can utilize brackets 
+        # it extracts the value in form of some stuff
+        # which has previously been enclosed with brackets and
+        # with a comma or end of line existing after the closing one
+        frames = dict()
+        for match_object in string.split(","):
+            frameParam = match_object.split(":")
+            max_f = self.max_frames -1
+            s = self.seed
+            frame = int(self.sanitize_value(frameParam[0])) if check_is_number(self.sanitize_value(frameParam[0].strip())) else int(numexpr.evaluate(frameParam[0].strip().replace("'","").replace('"',"")[::-1].replace("'","").replace('"',"")[::-1]))

```

Comment:
```suggestion
            frame = int(self.sanitize_value(frameParam[0])) if check_is_number(self.sanitize_value(frameParam[0].strip())) else int(numexpr.evaluate(frameParam[0].strip().replace("'","").replace('"',"")))
```
if we have a string and remove elements, we shouldn't need to do it again

Reply:
hmm, yes, I intended to remove them from the beginning and the end only, but forgot about removing the former check. Thank you!

Reply:
well, not removing, but setting the replace count to 1

Reply:
Added count=1 and tested it. Thanks!